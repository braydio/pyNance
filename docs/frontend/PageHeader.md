# PageHeader Component

Standard page header used at the top of views. Provides slots for an optional icon, title, subtitle, and an `actions` area on the right.

## Slots
- `icon` – optional leading icon
- `title` – main heading text
- `subtitle` – supporting description text
- `actions` – optional right‑aligned controls (e.g. buttons)

## Example
```vue
<PageHeader>
  <template #icon>
    <CreditCard class="w-6 h-6" />
  </template>
  <template #title>Transactions</template>
  <template #subtitle>View and manage your transactions</template>
  <template #actions>
    <UiButton variant="outline">Import</UiButton>
  </template>
</PageHeader>
```
