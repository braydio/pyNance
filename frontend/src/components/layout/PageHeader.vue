<template>
  <Card class="p-6 flex flex-col items-center text-center gap-2">
    <component v-if="icon" :is="icon" class="w-8 h-8" />
    <div>
      <h1 class="text-2xl font-bold">{{ title }}</h1>
      <p v-if="subtitle" class="text-muted">{{ subtitle }}</p>
    </div>
  </Card>
</template>

<script setup>
/**
 * PageHeader provides a consistent card-style header with optional icon and subtitle.
 */
import Card from '@/components/ui/Card.vue'

defineProps({
  /** Main heading text for the page */
  title: { type: String, required: true },
  /** Optional subtitle displayed below the title */
  subtitle: { type: String, default: '' },
  /** Optional icon component rendered above the title */
  icon: { type: [Object, Function], default: null },
})
</script>
