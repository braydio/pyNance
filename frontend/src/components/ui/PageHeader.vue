<template>
  <Card class="p-6 flex items-center">
    <div class="flex items-center gap-3 flex-1">
      <slot name="icon" />
      <div>
        <h1 class="text-2xl font-bold">
          <slot name="title" />
        </h1>
        <p class="text-muted">
          <slot name="subtitle" />
        </p>
      </div>
    </div>
    <div v-if="$slots.actions" class="flex items-center gap-2 ml-auto">
      <slot name="actions" />
    </div>
  </Card>
</template>

<script setup>
/**
 * PageHeader
 * Standard card header for views with an optional actions slot.
 *
 * Slots:
 * - icon: optional leading icon
 * - title: main heading text
 * - subtitle: supporting description text
 * - actions: optional right-aligned action controls
 */
import Card from '@/components/ui/Card.vue'
</script>
