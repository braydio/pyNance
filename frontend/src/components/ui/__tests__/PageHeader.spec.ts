// @vitest-environment jsdom
import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import PageHeader from '../PageHeader.vue'

describe('PageHeader', () => {
  it('renders title', () => {
    const wrapper = mount(PageHeader, {
      slots: { title: 'Dashboard' },
    })
    expect(wrapper.find('h1').text()).toBe('Dashboard')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders optional subtitle and icon', () => {
    const wrapper = mount(PageHeader, {
      slots: {
        title: 'Dashboard',
        subtitle: 'Overview',
        icon: '<span class="icon" />',
      },
    })
    expect(wrapper.find('p.text-muted').text()).toBe('Overview')
    expect(wrapper.find('.icon').exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders actions slot', () => {
    const wrapper = mount(PageHeader, {
      slots: {
        title: 'Dashboard',
        actions: '<button>Action</button>',
      },
    })
    expect(wrapper.find('div.flex.items-center.gap-2.ml-auto').exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
