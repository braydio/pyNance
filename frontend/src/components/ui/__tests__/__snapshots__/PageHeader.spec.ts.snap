// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeader > renders actions slot 1`] = `
"<div class="card glass p-6 flex items-center">
  <div class="flex items-center gap-3 flex-1">
    <div>
      <h1 class="text-2xl font-bold">Dashboard</h1>
      <p class="text-muted"></p>
    </div>
  </div>
  <div class="flex items-center gap-2 ml-auto"><button>Action</button></div>
</div>"
`;

exports[`PageHeader > renders optional subtitle and icon 1`] = `
"<div class="card glass p-6 flex items-center">
  <div class="flex items-center gap-3 flex-1"><span class="icon"></span>
    <div>
      <h1 class="text-2xl font-bold">Dashboard</h1>
      <p class="text-muted">Overview</p>
    </div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`PageHeader > renders title 1`] = `
"<div class="card glass p-6 flex items-center">
  <div class="flex items-center gap-3 flex-1">
    <div>
      <h1 class="text-2xl font-bold">Dashboard</h1>
      <p class="text-muted"></p>
    </div>
  </div>
  <!--v-if-->
</div>"
`;
